src/engine/strat_engine/udev.rs

```diff
@@ -157,10 +157,60 @@ where
 {
     let context = libudev::Context::new()?;
     let mut enumerator = block_enumerator(&context)?;
-
     Ok(enumerator
         .scan_devices()?
         .filter(|dev| dev.is_initialized())
         .find(|x| x.devnode().map_or(false, |d| **device_path == *d))
         .map(|ref d| f(&UdevEngineDevice::from(d))))
 }
+
+/// Return true if udev queue is empty and no events are being processed.
+#[cfg(test)]
+pub fn is_udev_queue_empty() -> StratisResult<bool> {
+    // Path of udev queue's file, typically this file is empty, it just asses that udev events are still
+    // being processed. When the file disappeard from the filesystem, the queue is empty
+    const UDEV_QUEUE_FILE_PATH: &str = "/run/udev/queue";
+
+    let udev_queue_file_path = &std::path::Path::new(UDEV_QUEUE_FILE_PATH);
+
+    // If the file exists then the queue is not empty and we return an error
+    if udev_queue_file_path.exists() {
+        Err(StratisError::Msg(format!(
+            "Udev queue file {} still exists",
+            UDEV_QUEUE_FILE_PATH
+        )))
+    } else {
+        Ok(true)
+    }
```

Comment:
Mainly stylistic, but I'd think this return type should either be `StratisResult<()>` or `bool`. I don't see any case where `Ok(false)` is returned.